sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -40,6 +41,46 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       .getOne();
   }
 
+  /**
+   * Gets a program details for a student
+   * This returns only a subset of the educationProgram
+   * id in the query.
+   * @param programId Program id.
+   * @returns program
+   */
+  async getStudentEducationProgram(
+    programId: number,
+  ): Promise<StudentEducationProgramSummary> {
+    const studentEducationProgram = await this.repo
+      .createQueryBuilder("programs")
+      .select([
+        "programs.id",
+        "programs.name",
+        "programs.description",
+        "programs.credentialType",
+        "programs.credentialTypeOther",
+        "programs.deliveredOnSite",
+        "programs.deliveredOnline",
+      ])
+      .where("programs.id = :programId", { programId })
```

Comment:
I believe that we should restrict this a bit more. Can the code do the check if at least the program is approved?
```
.where("programs.approvalStatus = :approvalStatus", {
        approvalStatus: ApprovalStatus.approved,
      })
```

Reply:
I agree, even for the list we are showing only the approved programs